registration/management/commands/cron_metrics.py

```diff
@@ -0,0 +1,62 @@
+from django.core.management.base import BaseCommand, CommandError
+from django.db.models import Q
+
+from registration.models import *
+
+
+"""
+There's a few options for where these metrics could end up.  These should eventually be configurable:
+ - built-in database metrics table: (simplest, no external dependancy).
+ - prometheus: Time-series database, with existing reporters for e.g. django request processors.
+ - influxdb: Another great time-series database.
+"""
+
+
+def get_paid_order_items(event):
+    """
+    Returns all order items for paid and completed orders with a valid price-level assignment.
+    (Effectively, the total number of badges for a particular event)
+
+    :param event: event to filter by
+    :return: List[order_items]
+    """
+    order_items = OrderItem.objects.filter(badge__event=event).exclude(
+        Q(order__isnull=True)
+        | Q(order__billingType=Order.UNPAID)
+        | Q(order__status__in=(Order.FAILED, Order.REFUNDED, Order.REFUND_PENDING))
+        | Q(priceLevel__isnull=True)
+    )
+    return order_items
+
+
+class Command(BaseCommand):
+    help = "Calculates metrics for all events to record to time-series"
+
+    def handle(self, *args, **options):
+        for event in Event.objects.filter():
+            # Attendee data
+            order_items = get_paid_order_items(event)
+            total_count = order_items.count()
+
+            # Bin by registration level
+            level_bins = {}
+            price_levels = PriceLevel.objects.filter()
+            for level in price_levels:
+                level_bins[level.id] = order_items.filter(priceLevel=level).count()
+
+            # Staff counts
+            staff = Staff.objects.filter(event=event)
+            total_staff_count = staff.count()
+
+            active_staff_count = 0
+            for s in staff:
+                if s.attendee and s.attendee.badge_set.filter().count():
```

Comment:
This is slooooooow - kicks off O(2n) queries for every staff in this comprehension, need to figure out how to get the ORM to use joins instead.

Reply:
Resolved with `prefetch_related()` chaining in ORM.